### PR TITLE
Remove request pattern from tests using dynamic URLs

### PR DESCRIPTION
**Pull request by kian-thompson**
The `request` pattern has been deprecated and all out tests need to switch to the new `entryPoint` pattern or the `resolveHandle` method.

Continuation of https://github.com/microsoft/FluidFramework/pull/18365

[AB#6303](https://dev.azure.com/fluidframework/internal/_workitems/edit/6303)

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts [hunk 1]
```diff
@@ -630,14 +599,16 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
...
-						container2,
-						"/",
-					);
+					const defaultDataStoreContainer2 =
+						(await container2.getEntryPoint()) as ITestDataObject;
 					defaultDataStoreContainer2._root.set("mode", "write");
 					await waitForContainerWriteModeConnectionWrite(container2);
 
-					const response = await containerRuntime_resolveHandle(container2, {
+					const entryPoint = (await container2.getEntryPoint()) as ITestDataObject;
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts:607]: nit: You can use `defaultDataStoreContainer2` here instead of getting the entry point again.

File: packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts [hunk 2]
```diff
@@ -155,10 +154,11 @@ describeNoCompat("GC data store sweep tests", (getTestObjectProvider) => {
 			summaryVersion,
 		);
 
-		const summarizerDataObject = await requestFluidObject<ITestDataObject>(
-			summarizingContainer2,
-			testDataObject.handle.absolutePath,
-		);
+		const containerRuntime = (summarizer2 as any).runtime as ContainerRuntime;
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcSweepDataStores.spec.ts:157]: FYI: I am not 100% sure if we should keep all usages of resolveHandle in GC tests. I created a task for the GC crew to look into this - AB#6332

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts [hunk 3]
```diff
@@ -218,7 +224,9 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
 				// Load a container from the above summary. Retrieving the blob via any of the handles should fail. Note
 				// that the blob is requested via its url since this container does not have access to the blob's handle.
 				const container2 = await loadContainer(summary2.summaryVersion);
-				const response1 = await container2.request({ url: blobHandle1.absolutePath });
+				const entryPoint = (await container2.getEntryPoint()) as ITestDataObject;
+				const runtime = entryPoint._context.containerRuntime as ContainerRuntime;
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts:228]: nit: runtime2

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts [hunk 4]
```diff
@@ -361,7 +373,9 @@ describeNoCompat("GC attachment blob tombstone tests", (getTestObjectProvider) =
 
 				// Retrieving the blob should fail. Note that the blob is requested via its url since this container does
 				// not have access to the blob's handle.
-				const response = await container2.request({ url: blobHandle.absolutePath });
+				const entryPoint = (await container2.getEntryPoint()) as ITestDataObject;
+				const runtime = entryPoint._context.containerRuntime as ContainerRuntime;
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneAttachmentBlobs.spec.ts:377]: nit: runtime2 everywhere its for container2

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts [hunk 5]
```diff
@@ -787,31 +802,46 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
 					url: unreferencedId,
 					headers: { [AllowTombstoneRequestHeaderKey]: true },
 				};
-				const tombstonedObject = await requestFluidObject<ITestDataObject>(
-					tombstoneContainer,
-					requestAllowingTombstone,
-				);
+				let tombstonedObject: ITestDataObject;
+				{
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:806]: Why is this (and similar code below) in a separate block?

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts [hunk 6]
```diff
@@ -787,31 +802,46 @@ describeNoCompat("GC data store tombstone tests", (getTestObjectProvider) => {
...
-					sendingContainer,
-					unreferencedId,
-				);
+				let sendDataObject: ITestDataObject;
+				{
+					const sendingEntryPoint =
+						(await sendingContainer.getEntryPoint()) as ITestDataObject;
+					const sendingContainerRuntime = sendingEntryPoint._context
+						.containerRuntime as ContainerRuntime;
+					const response = await sendingContainerRuntime.resolveHandle({
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:840]: nit: Use `resolveHandleHelper` here? Maybe you can udpate `resolveHandleHelper` to accept the container instead of container runtime and move the logic there.

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:840]: I see, you are also using it for the summarizer. never mind.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 7]
```diff
@@ -837,8 +837,11 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		await waitForContainerConnection(container2);
 
 		// get new datastore from first container
-		const dataStore2 = await requestFluidObject<ITestFluidObject>(container1, id);
-		const map2 = await requestFluidObject<SharedMap>(dataStore2.runtime, newMapId);
+		const entryPoint = (await container1.getEntryPoint()) as ITestFluidObject;
+		const containerRuntime = entryPoint.context.containerRuntime as ContainerRuntime;
+
+		const response = await containerRuntime.resolveHandle({ url: `/${id}/${newMapId}` });
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:843]: I don't think this should be using resolveHandle. It's supposed to be used only for back-compat. You could get the entryPoint (defaultDataStore) from container2 and get the dataStore2's handle via `defaultDataStore.root.get("someDataStore")`. Maybe put "someDataStore" in a variable outside the pendingOps block.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts [hunk 8]
```diff
@@ -887,7 +890,10 @@ describeNoCompat("stashed ops", (getTestObjectProvider) => {
 		// get new DDS from first container
 		await provider.ensureSynchronized();
 		const dataStore1 = (await container1.getEntryPoint()) as ITestFluidObject;
-		const map2 = await requestFluidObject<SharedMap>(dataStore1.runtime, newMapId);
+		const containerRuntime = dataStore1.context.containerRuntime as ContainerRuntime;
+
+		const response = await containerRuntime.resolveHandle({ url: `/default/${newMapId}` });
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:895]: Same as above, this should not be using resolveHandle. You can do `dataStore1.runtime.getChannel(newMapId)` to get the map.

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:895]: Same for the rest of the usages in this file.

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 9]
```diff
@@ -326,67 +323,57 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
...
-				ContainerRuntimeFactoryWithDefaultDataStore,
-				registryStoreEntries,
-			);
-			await assert.doesNotReject(
-				summarizeNow(createSummarizerResult.summarizer, "test"),
-				"Summarizing should not throw",
-			);
+		// Create a container for the first client.
+		const container1 = await provider.createContainer(runtimeFactory);
+		await container1.getEntryPoint();
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:343]: Should't this be wrapped in an `assert.doesNotReject` to keep the behavior same as before?

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 10]
```diff
@@ -326,67 +323,57 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
...
 
-			// In summarizer, load the data store should fail.
-			await assert.rejects(
-				requestFluidObject<TestDataObject1>(createSummarizerResult.container, "/"),
-				(e: Error) =>
-					e.message ===
-					"Non interactive/summarizer client's data object should not be initialized",
+		// In summarizer, load the data store should fail.
+		try {
+			const runtime = (createSummarizerResult.summarizer as any).runtime as ContainerRuntime;
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:361]: nit: Maybe put these 3 lines in a const block and use assert.rejects? It just looks cleaner.

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts [hunk 11]
```diff
@@ -326,67 +323,57 @@ describeNoCompat("Summaries", (getTestObjectProvider) => {
...
-			const container2 = await provider.loadContainer(runtimeFactory);
-			await assert.doesNotReject(
-				requestFluidObject<TestDataObject1>(container2, "/"),
-				"Loading data store in second interactive client should not throw.",
-			);
-		},
-	);
+		// Load second container, load the data store will also call initializingFromExisting and succeed.
+		const container2 = await provider.loadContainer(runtimeFactory);
+		await container2.getEntryPoint();
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:375]: Same comment as above. Should't you keep the `assert.doesNotReject`?

File: packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts [hunk 12]
```diff
@@ -648,8 +662,10 @@ describeFullCompat(`Dehydrate Rehydrate Container Test`, (getTestObjectProvider)
 					await createDetachedContainerAndGetEntryPoint();
 
 				// Create and reference another dataStore
+				const dataStore2Alias = "dataStore2";
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts:665]: Same comment for this file as this one - https://github.com/microsoft/FluidFramework/pull/18427/files?show-viewed-files=true&file-filters%5B%5D=#r1402487874

File: packages/test/test-service-load/src/loadTestDataStore.ts [hunk 13]
```diff
@@ -129,7 +129,9 @@ export class LoadTestDataStoreModel {
 		// If we did not create the data store above, load it by getting its url.
 		if (gcDataStore === undefined) {
 			const gcDataStoreId = root.get(gcDataStoreIdKey);
-			const response = await containerRuntime.request({ url: `/${gcDataStoreId}` });
+			const response = await (containerRuntime as ContainerRuntime).resolveHandle({
```
Comment (agarwal-navin) [inline packages/test/test-service-load/src/loadTestDataStore.ts:132]: These tests should not be using resolveHandle. I created a task to re-visit this AB#6334

Comment (agarwal-navin) [reply packages/test/test-service-load/src/loadTestDataStore.ts:132]: Not blocking for this PR of course.

File: packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts (hunk 12 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts:665]: This seems like a "nit" type suggestion. Is one pattern better than the other in your eyes? (entryPoint vs retrieving the handle from root data object)

Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/deRehydrateContainerTests.spec.ts:665]: It's not about one pattern being better than the other. It's more about what the intention of the test is. For example, a lot of the GC tests do not want to create root data stores because they want to be able to unreference them later and you can't unreference root data stores.

I don't know what the intention is for these data stores so I suggested keeping the behavior the same. You can maybe change it and create a work item for the test owner to ensure the test is still doing whats expected.

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts (hunk 5 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:806]: Was just a style choice to re-use the `response` name. I have no strong opinion one way or the other

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts (hunk 6 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts:840]: I created that helper strictly to retain the testing around `assert.throws` and `assert.doesNotThrow`. It provides no value for this case

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 7 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:843]: Wouldn't getting `defaultDataStore` from `container2` change what this test is doing? Also, if I use `container1` for getting the entryPoint I need to move the `ensureSynchronized()` call above reaching into `defaultDataStore.root` as the handle won't be there.

File: packages/test/test-end-to-end-tests/src/test/summaries.spec.ts (hunk 9 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/summaries.spec.ts:343]: Not sure what happened before. When I first fixed these tests, it didn't like the `rejects/doesNotReject` calls.

File: packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts (hunk 7 above)
Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:843]: Maybe you are right, I am not sure. I would defer to the test owner to find a way to implement this without resolveHandle unless they explicitly want to use it for some reason. Maybe tag them here or create a follow up task.

Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/stashedOps.spec.ts:843]: [AB#6340](https://dev.azure.com/fluidframework/internal/_workitems/edit/6340)

File: packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts [hunk 14]
```diff
@@ -70,7 +64,6 @@ describeNoCompat("GC inactive nodes tests", (getTestObjectProvider) => {
 		const { throwOnInactiveLoad } = params;
 		return {
 			runtimeOptions: {
-				summaryOptions: { summaryConfigOverrides: { state: "disabled" } },
```
Comment (agarwal-navin) [inline packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts:73]: Why is this removed?

File: packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts (hunk 5 above)
Comment (agarwal-navin) [reply packages/test/test-end-to-end-tests/src/test/gc/gcTombstoneDataStores.spec.ts]: Me neither :-). I was just curious.

**Review by agarwal-navin — APPROVED** [review]

File: packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts (hunk 14 above)
Comment (kian-thompson) [reply packages/test/test-end-to-end-tests/src/test/gc/gcInactiveNodes.spec.ts:73]: A lot of tests are creating summarizer containers, and `Summarizer` objects won't exist when you have `state: "disabled"`. This is a problem when you try to get default data objects from the summarizer container as it'll just hit assert `0x5bf /* Summarizer object is undefined in a summarizer client */`
